### PR TITLE
Add number type

### PR DESCRIPTION
**Pull request by alexkwolfe**

### CODE REVIEW

File: src/number.coffee [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    if _.isNumber(string)
+      new Number(string)
+    else
+      sanitized = string.replace(/[^\d.]/g, '')
+      if sanitized.length
+        new Number(sanitized)
+      else
+        new Number(string)
+
+  number.valid = !isNaN(number)
```
Comment (cgrayson) [inline src/number.coffee]: Did you already take in to account the [confusing special-case behavior](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/isNaN#Confusing_special-case_behavior) of `isNan()`?

Comment (alexkwolfe) [inline src/number.coffee]: I think so. In this implementation `isNaN()` is always called with a `Number` instance. Let me add a few more tests to make sure.

**Comment by cgrayson** [comment]
This still parses whitespace as valid, with numeric value 0.

**Comment by cgrayson** [comment]
LGTM

**Pull request merged**